### ISSUE TITLE
borked ESS install (Ubuntu 16.04/Emacs 24.5.1); how to get back?

### ISSUE THREAD

**Issue by bbolker** [emacs-ess/ESS#632]
On Ubuntu 16.04 (actually Xubuntu)/Emacs version 24.5.1, I seem to have screwed up my ESS installation and can't restore it ...

` sudo apt install ess` gives:

> Install ess for emacs24
install/ess: Handling install for emacsen flavor emacs24
ERROR: install script from ess package failed
dpkg: error processing package ess (--configure):
 subprocess installed post-installation script returned error exit status 1
Errors were encountered while processing:
 ess
E: Sub-process /usr/bin/dpkg returned an error code (1)

   I was able to find the post-installation script and run it by hand:

```
 sudo /usr/lib/emacsen-common/emacs-package-install --postinst ess
```

  this replicates the error (listed as `ERROR: ... `) above.

  But I don't know where to look for the error logs for this step to do further debugging.

  If I try to install through emacs via MELPA, i.e. "M-x package-install ess", I get: 

> https://melpa.org/packages/julia-mode-20160311.1351.el: Not found

  (This is with a stripped-down .emacs file that *only* includes the stuff that enables package installation ...)

Where should I look next for diagnostics/debugging?

**Comment by eddelbuettel** [comment]
*tl;dr:* The postint and prerm scripts live in `/var/lib/dpkg/info/ess.*`.  When things are really sour I sometimes edit them (as root) and put a `exit 0` at the top so that they do not get in the way.

With that, you should be able to do `sudo apt-get purge ess` after which you should be able to reinstall.

Coincidentally, I spent this weekend on turning this into `ess` (for transition) and `elpa-ess` allowing use along with elpa etc.  I could offer you those (they are currently in Debian's review queue as `elpa-ess` triggers a 'oh, look, a new package' step that requires a manual review / tab on the back) but I only tested these on my 18.04 system and I am not sure if they work on something older.

**Comment by bbolker** [comment]
Never mind; I read #629 and switched to melpa "bleeding edge" and managed to install that way.

I'm still curious what's wrong with the apt-get route, but it's probably not worth the trouble ...

**Comment by eddelbuettel** [comment]
See my answer.

(I also don't understand why people insist on stability (yo, 16.04 is outdated, use 18.04) yet jump to bleeding edge source repos but I guess I digress).

Also, filing _upstream_ tickets is not quite the right thing for _distro_ bugs...  But hey, you found me.  By spamming all upstream maintainers.

**Comment by eddelbuettel** [comment]
> Emacs version 24.5.1

That is probably too old for really new versions.  Not sure what melpa-stable has.  At work, on (ugghh...) CentOS I built Emacs 26.1 from source and go straight to melpa etc.

**Comment by bbolker** [comment]
(1) Thanks. (Our comments keep crossing.)
(2) I don't love stability, I just hate taking the time to update and deal with the inevitable fallout. (It's never perfect.) You might laugh, or groan, if you new how recently I'd updated from 14.04 ...
(3) The reason I went to bleeding-edge is #629 (where @vspinu says "AFAICS melpa stable is not even recommended by melpa maintainers")  - and it did work ...
(4) TBH I didn't *know* what kind of a problem I have (i.e. whether I have a distro problem or something else); the fact that I initially failed through two separate channels made me think I might have a broader problem. Here at least I thought I would find people who knew how to diagnose ESS problems ...

**Comment by eddelbuettel** [comment]
No, all good, and happy you're up and running.  But the ESS'ers would not have know about `/var/lib/dpkg/info/ess.prerm` failing for you ...

Anyway, all good now.

**Comment by molberg** [comment]
I had the same problem on a Gallium OS 16.04 (Ubuntu for chroembooks). I did a

> sudo emacs -nw /usr/share/emacs/site-lisp/ess/ess-install.el

followed by eva-buffer from within emacs. It produced a number of warnings but seems to have solved it for me.

**Comment by eddelbuettel** [comment]
@molberg Not the best approach, see the thread starting at https://stat.ethz.ch/pipermail/r-sig-debian/2018-August/002941.html
